### DISCUSSION TITLE
The json delete method in PHPUnit does not work

### DISCUSSION THREAD

**Discussion by thomas-tahmasebi** [laravel/framework#45408] [Q&A]
- Laravel Version: 9.19
- PHP Version: 8.1.13
- Database Driver & Version:

### Description:
In PHPunit, I would like to send a delete request using a json method as follows:
`$this->json('delete', '/api/roles/99', [])`
But it will return this:

> 1) Tests\Unit\RoleTest::test_delete_a_role
Error: Call to undefined method Tests\Unit\RoleTest::json()

/var/www/html/tests/Unit/RoleTest.php:16

ERRORS!
Tests: 2, Assertions: 1, Errors: 1.

**Comment by johanrosenson** [comment]
unit tests extend `PHPUnit\Framework\TestCase` by default, you need to extend `Tests\TestCase` instead to get the http test helpers.

**Reply by thomas-tahmasebi** [reply]
I generated this file by `php artisan make:test --unit`, So why should i change it?

**Reply by johanrosenson** [reply]
Because you want it to work ;)

Actually you probably wanted to create a feature test instead of a unit test if you need to do http requests. Then these methods will be available for you without needing any changed.

**Reply by thomas-tahmasebi** [reply]
Sorry i changed my comment, I mean it should extend `Tests\TestCase` by default when i'm using that command. Am i right?

**Reply by johanrosenson** [reply]
No, only the feature tests extend that by default.

The unit tests are meant to test smaller parts of your app, and you usually don't need to make HTTP requests in a unit test, for that you are recommended to use a feature test.

**Reply by johanrosenson** [reply]
If you want you can modify the built in template for a unit test to always extend it.

Use `php artisan stub:publish` to publish all the built-in stubs, then edit the `stubs/test.unit.stub` file to your liking.
